developer-guide/development/quickstart.md

```diff
@@ -0,0 +1,248 @@
+
+> Note: KubeSphere uses `go module` to manage dependencies, but the development process still relies on `GOPATH`
+> In the CRD development process, you need to use tools to automatically generate code. The tools used by KubeSphere still need to rely on `GOPATH`.
+> For Chinese contributors who are going to pull the go module, we recommend you to use [goproxy.cn](https://goproxy.cn) as the proxy.
+
+### Kubesphere
+
+See [how to install kubesphere](https://kubesphere.io/docs/).
+
+## Hello KubeSphere
+```bash
+git clone https://github.com/kubesphere/kubesphere.git
+cd kubesphere && mkdir -p kapis/hellokubesphere/v1alpha1 && cd kapis/hellokubesphere/v1alpha1
+```
+Create a file named hander.go. In this tutorial, it only returned a http response with "hello-kubesphere", 
+you can also write your custom logic.  
+```bash
+package hellokubesphere
+
+import "github.com/emicklei/go-restful"
+
+type handler struct {}
+
+func newHandler() handler{
+	return handler{}
+}
+
+func (h handler) HelloKubeSphere (req *restful.Request, resp *restful.Response) {
+	resp.WriteAsJson(HelloResponse{
+		Message: "hello kubesphere",
+	})
+}
+
+type HelloResponse struct {
+	Message string `json:"message"`
+}
+```
+Once you have completed your logic part, you need to design your api path and http method and so on with a file called register.go.
+See more details on [go-restful](https://github.com/emicklei/go-restful).
+```bash
+package hellokubesphere
+
+import (
+	"net/http"
+
+	"github.com/emicklei/go-restful"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"kubesphere.io/kubesphere/pkg/api"
+	"kubesphere.io/kubesphere/pkg/apiserver/runtime"
+)
+
+const (
+	GroupName = "example.kubesphere.io"
+)
+
+var GroupVersion = schema.GroupVersion{Group: GroupName, Version: "v1alpha1"}
+
+func AddToContainer(container *restful.Container) error {
+	webservice := runtime.NewWebService(GroupVersion)
+	handler := newHandler()
+
+	webservice.Route(webservice.GET("/hello-kubesphere").
+		Reads("").
+		To(handler.HelloKubeSphere).
+		Returns(http.StatusOK, api.StatusOK, HelloResponse{})).
+		Doc("Api for hello-kubesphere")
+
+	container.Add(webservice)
+
+	return nil
+}
+```
+Cause you have created a folder named v1alpha1, you need to set Version to v1alpha1.
+See more details on [GVK](https://book.kubebuilder.io/cronjob-tutorial/gvks.html).<br>
+Once you have created those two files, you still need to register your api into go-restful.
+Go back kubesphere root dir and edit pkg/apiserver/apiserver.go.<br>
+Put 
+
+```bash
+urlruntime.Must(hellokubesphere.AddToContainer(s.container))
+```
+at the end of installKubeSphereAPIs function. <br>
+Kubesphere has strict access control, for convenience, you can modify the buildHandlerChain function in  pkg/apiserver/apiserver.go  and change excludedPaths variable like this
+
+```
+excludedPaths := []string{"/oauth/*", "/kapis/config.kubesphere.io/*", "/kapis/version", "/kapis/example.kubesphere.io/v1alpha1/hello-kubesphere"}
```

Comment:
Don't add example to exclueded api pathes, it's only authentication and authorization apis